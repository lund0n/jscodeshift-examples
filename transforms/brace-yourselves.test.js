/* eslint-env jest */
const { createTransformer, readInput, readOutput } = require('./test-utils');
const braceYourselves = require('./brace-yourselves');

const transform = createTransformer(braceYourselves);

describe('using snapshots', () => {
  test('wraps the if in braces', () => {
    const input = `if (true) return "Yes"`;
    const output = transform(input);
    expect(output).toMatchSnapshot();
  });

  test('ignores ifs that already have braces', () => {
    const input = `if (true) { return "Yes"}`;
    const output = transform(input);
    expect(output).toMatchSnapshot();
  });

  test('ignores elses that already have braces', () => {
    const input = `if (true) { return "Yes"} else { return "No"}`;
    const output = transform(input);
    expect(output).toMatchSnapshot();
  });
});

describe('using inline snapshots', () => {
  test('wraps the else in braces', () => {
    const input = `
if (true) {
  return "Yes"
} else return "No"
`;
    const output = transform(input);
    expect(output).toMatchInlineSnapshot(`
"
if (true) {
  return \\"Yes\\"
} else {
  return \\"No\\"
}
"
`);
  });
});

describe('using fixtures', () => {
  const input = readInput('brace-yourselves');
  const expected = readOutput('brace-yourselves');
  const output = transform(input);
  expect(output).toEqual(expected);
});
