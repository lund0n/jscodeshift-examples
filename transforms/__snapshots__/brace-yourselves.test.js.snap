// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`using snapshots ignores elses that already have braces 1`] = `"if (true) { return \\"Yes\\"} else { return \\"No\\"}"`;

exports[`using snapshots ignores ifs that already have braces 1`] = `"if (true) { return \\"Yes\\"}"`;

exports[`using snapshots wraps the if in braces 1`] = `
"if (true) {
  return \\"Yes\\"
}"
`;
